docset/winserver2012-ps/kds/Add-KdsRootKey.md

```diff
@@ -33,28 +33,35 @@ The Microsoft Group KdsSvc generates new group keys from the new root key.
 ## EXAMPLES
 
 ### Example 1: Generate a new root key
-```
+```powershell
 PS C:\>Add-KdsRootKey
 ```
 
 This example generates a new root key for the Microsoft Group KdsSvc within AD.
 
 ### Example 2: Generate a new root key for immediate use
-```
+```powershell
 PS C:\>Add-KdsRootKey -EffectiveImmediately
 ```
 
-This example generates a new root key immediately and adds it to the Microsoft Group KdsSvc.
+This example generates a new root key and will be usable after 10 hours of creation and adds it to the Microsoft Group KdsSvc.
+>[!TIP] This inmediate use is conditioned to wait 10 hours from creation time to allow all DCs to converge AD replication befure you can create gMSA. This time prevents password generation from occurring before all DCs in environment capable of answering gMSA request.>
+
+>[!TIP] For inmediate use please use:
+>```Powershell 
+>Add-KdsRootKey �EffectiveTime ((get-date).addhours(-10))  
+>``` 
+>*Allows using gMSAs **immediately**, because it sets the start time 10 hours in past.*
```

Comment:
This allows the use of gMSAs immediately because it sets the start time back by 10 hours.

Reply:
Johan please improve this, not certain

Reply:
Agree.